app/src/org/commcare/activities/EntitySelectActivity.java

```diff
@@ -872,8 +941,8 @@ public void deliverLoadResult(List<Entity<TreeReference>> entities,
 
         findViewById(R.id.entity_select_loading).setVisibility(View.GONE);
 
-        if (adapter != null && filterString != null && !"".equals(filterString)) {
-            adapter.applyFilter(filterString);
+        if (adapter != null) {
+            restoreAdapterStateFromSession();

```

Comment:
Can you explain what this is doing / what the circumstances are that necessitate it?